modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,445 @@
+                        c[i] = c[i] + 1;
+                    }
+                    else
+                    {
+                        c[i] <<= 2;
+                    }
+                }
+            }
+        };
+        //!A madified cs census that compares a pixel with the imediat neightbour starting
+        //!from the center
+        struct ModifiedCsCensus
+        {
+            uint8_t **image;
+            int n2;
+            int imageStop;
+            ModifiedCsCensus(){}
+            ModifiedCsCensus(uint8_t **images, int ker, int numberOfImages = CV_DUAL_DESCRIPTOR)
+            {
+                image = new uint8_t *[numberOfImages];
+                for(int i = 0; i < numberOfImages; i++)
+                    image[i] = images[i];
+                imageStop = numberOfImages;
+                n2 = ker;
+            }
+            void operator()(int rrWidth,int w2, int rWidth, int jj, int j, int c[20]) const
+            {
+                (void)j;
+                (void)rWidth;
+                for(int i = 0; i < imageStop; i++)
+                {
+                    if (image[i][(rrWidth + jj)] > image[i][(w2 + (jj + n2))])
+                    {
+                        c[i] = c[i] + 1;
+                    }
+                    c[i] = c[i] * 2;
+                }
+            }
+        };
+        //!A kernel in which a pixel is compared with the center of the window
+        struct CensusKernel
+        {
+            uint8_t **image;
+            int imageStop;
+            CensusKernel(){}
+            CensusKernel(uint8_t **images, int numberOfImages = CV_DUAL_DESCRIPTOR)
+            {
+                image = new uint8_t *[numberOfImages];
+                for(int i = 0; i < numberOfImages; i++)
+                    image[i] = images[i];
+                imageStop = numberOfImages;
+            }
+            void operator()(int rrWidth,int w2, int rWidth, int jj, int j, int c[20]) const
+            {
+                (void)w2;
+                for(int i = 0; i < imageStop; i++)
+                {
+                    ////compare a pixel with the center from the kernel
+                    if (image[i][rrWidth + jj] > image[i][rWidth + j])
+                    {
+                        c[i] += 1;
+                    }
+                    c[i] <<= 1;
+                }
+            }
+        };
+        //template clas which efficiently combines the descriptors
+        template <int step_start, int step_end, int step_inc,int nr_img, typename Kernel>
+        class CombinedDescriptor:public ParallelLoopBody
+        {
+        private:
+            int width, height,n2;
+            int **dst;
+            Kernel kernel_;
+            int n2_stop;
+        public:
+            CombinedDescriptor(int w, int h, int k2, int **distance, Kernel kernel,int k2Stop)
+            {
+                width = w;
+                height = h;
+                n2 = k2;
+                dst = new int*[nr_img];
```

Comment:
Let's stick with a static-sized array. As far as I see, there's no need for heap allocations.